<launch>
    <arg name="robot_name" default="robot1"/>

    <arg name="map_frame_id" default="map"/>
    <arg name="pose_topic" default="pose"/>
    <arg name="pose_log_topic" default="pose_log"/>
    <arg name="goal_topic" default="goal"/>
    <arg name="goal_list_topic" default="goal_list"/>
    <arg name="to_goal_topic" default="to_goal"/>
    <arg name="move_angle_topic" default="move_angle"/>
    <arg name="visualization_topic" default="visualization"/>

    <group ns="$(arg robot_name)">
        <node pkg="exploration" type="sensor_based_exploration" name="sensor_based_exploration">
            <remap from="scan" to="scan"/>
            <remap from="bumper" to="bumper"/>
            <remap from="velocity" to="velocity"/>

            <param name="map_frame_id" value="$(arg map_frame_id)"/>
            <remap from="pose" to="$(arg pose_topic)"/>
            <remap from="pose_log" to="$(arg pose_log_topic)"/>
            <remap from="goal" to="$(arg goal_topic)"/>
            <remap from="goal_list" to="$(arg goal_list_topic)"/>
            <remap from="to_goal" to="$(arg to_goal_topic)"/>
            <remap from="move_angle" to="$(arg move_angle_topic)"/>
        </node>

        <node pkg="exploration" type="visualization" name="visualization">
            <param name="exploration_method" value="SENSOR"/>

            <param name="map_frame_id" value="$(arg map_frame_id)"/>
            <remap from="pose" to="$(arg pose_topic)"/>
            <remap from="goal" to="$(arg goal_topic)"/>
            <remap from="goal_list" to="$(arg goal_list_topic"/>
            <remap from="to_goal" to="$(arg to_goal_topic)"/>
            <remap from="move_angle" to="$(arg move_angle_topic)"/>
            <remap from="pose_marker" to="$(arg visualization_topic)"/>
            <remap from="goal_marker" to="$(arg visualization_topic)"/>
            <remap from="goal_list_marker" to="$(arg visualization_topic)"/>
            <remap from="to_goal_marker" to="$(arg visualization_topic)"/>
            <remap from="move_angle_marker" to="$(arg visualization_topic)"/>
        </node>

        <node pkg="exploration" type="log" name="">
            <remap from="pose" to="$(arg pose_topic)"/>
            <remap from="pose_log" to="$(arg pose_log_topic)"/>
        </node>

    </group>
</launch>